Add kitti dataset

Sirokujira opened the pull request:
Issue #139 

use package : 
pykitti package(0.2.4/0.3.0)
parseTrackletXML.py
(Methods for parsing tracklets (e.g. dataset labels), 
 originally created by Christian Herdtweck.)

Function :
Display labels and boxes

not implemented
Grayscale processing

yuyu2172 commented:
Hi! Thank you for your PR! Some comments.

1. Could you change default data directory from `KITTI` to `kitti`? (similarly for COCO, we name the directory as `coco`)
2. Could you raise more friendly error when `date` or `drive_num` are not specified? (`label_resolution` option of `CityscapesSemanticSegmentationDataset` is a good example for situation like this)
3. Could you change the shape of `label` to `(0,)` when there is only one instance?
https://github.com/chainer/chainercv/blob/master/chainercv/utils/testing/assertions/assert_is_bbox_dataset.py#L51
4. Is it possible to drop support for gray scale image? If we drop support, can we use `pykitti==0.3.0` for all settings? I personally prefer to set `0.3.0` as the recommended pykitti version.

yuyu2172 commented:
Do you know any experiments in academic papers that can be supported by this dataset?

Sirokujira commented:
1. current data folder name Lower case(and download destination folder)

2. add check logic (check date, drivenum)

3. fixed code(label shape 2 dim -> 1 dim)

4. fixed use pykitti >= 0.3.0 (dropped function grayscale image process.)

>Do you know any experiments in academic papers that can be supported by this dataset?

source of a quote?
It is the item "raw dataset" of "Citation" at http://www.cvlibs.net/datasets/kitti/.
(add comment kitti_bbox_dataset.py)

I do not know about the research citing the data set.

yuyu2172 commented on chainercv/datasets/kitti/kitti_bbox_dataset.py:
```diff
@@ -0,0 +1,219 @@
+
+    When queried by an index, if :obj:`tracklet == True`,
+    this dataset returns a corresponding
+    :obj:`img, bbox, label`, a tuple of an image, bounding boxes, labels.
```
Could you make two classes: `KITTIBboxDataset` and `KITTIImageDataset`?
It is confusing that when `tracklet=False`, the dataset never returns unempty bounding box or label even though it is named as `BboxDataset`.
`KITTIBboxDataset` should only support `2011_09_26`, and `KITTIImageDataset` should support all the other dates.
Also, if you don't want to, you do not need to implement `KITTIImageDataset`.

We have two classes for Cityscapes as well because subset of the dataset does not contain annotation.

yuyu2172 commented on chainercv/datasets/__init__.py:
```diff
@@ -19,6 +19,13 @@
+from chainercv.datasets.kitti.kitti_bbox_dataset import KITTIBboxDataset  # NOQA
+from chainercv.datasets.kitti.kitti_utils import kitti_bbox_label_colors  # NOQA
+from chainercv.datasets.kitti.kitti_utils import kitti_bbox_label_names  # NOQA
+from chainercv.datasets.kitti.kitti_utils import kitti_date_lists  # NOQA
```
Do we need to expose this object at this namespace?
We only expose `_label_names` and `_colors` here.

yuyu2172 commented on chainercv/datasets/__init__.py:
```diff
@@ -19,6 +19,13 @@
+from chainercv.datasets.kitti.kitti_utils import kitti_bbox_label_colors  # NOQA
+from chainercv.datasets.kitti.kitti_utils import kitti_bbox_label_names  # NOQA
+from chainercv.datasets.kitti.kitti_utils import kitti_date_lists  # NOQA
+from chainercv.datasets.kitti.kitti_utils import kitti_date_num_dicts  # NOQA
```
ditto
